### PR TITLE
Add rollOption and blue light "Torch" for Crystal Luminescence feat

### PR DESCRIPTION
**Pull request by farling42**
This adds rules to the "Crystal Luminescence" feat for the Kashrishi race.

It provides a rollOption to enable/disable the light (20 ft bright, +20 ft dim) with a a blue colour to it.

It also provides the extra toggle for the second paragraph increasing the range under certain conditions.

### CODE REVIEW

File: packs/feats/crystal-luminescence.json [hunk 1]
```diff
@@ -25,7 +25,41 @@
...
+                    },
+                    "bright": "20+@item.flags.pf2e.rulesSelections.crystalLuminescence",
+                    "color": "#1e0ab8",
+                    "dim": "40+@item.flags.pf2e.rulesSelections.crystalLuminescence"
+                }
+            },
+            {
+                "domain": "all",
+                "key": "RollOption",
+                "label": "Crystal Luminescence",
```
Comment (simonward) [inline packs/feats/crystal-luminescence.json:46]: This line isn't necessary, as it should inherit the label from the item.

File: packs/feats/crystal-luminescence.json [hunk 2]
```diff
@@ -25,7 +25,41 @@
...
+                "option": "crystal-luminescence",
+                "selection": "0",
+                "suboptions": [
+                    {
+                        "label": "Normal",
```
Comment (simonward) [inline packs/feats/crystal-luminescence.json:51]: These labels should be localized.

**Comment by farling42** [comment]
See updated commits to resolve your two comments.

File: packs/feats/crystal-luminescence.json [hunk 3]
```diff
@@ -25,7 +25,40 @@
...
+                "key": "TokenLight",
+                "predicate": [
+                    "crystal-luminescence"
+                ],
+                "value": {
+                    "animation": {
+                        "type": "torch"
+                    },
+                    "bright": "20+@item.flags.pf2e.rulesSelections.crystalLuminescence",
+                    "color": "#1e0ab8",
```
Comment (TikaelSol) [inline packs/feats/crystal-luminescence.json:39]: We should give people the choice of color when taking the feat. See the rule elements on Effect: Spirit Power (Passion)

If we need a new color added to the system standard colors we should also add it to that effect and Lantern Surki.

File: packs/feats/crystal-luminescence.json [hunk 4]
```diff
@@ -25,7 +25,40 @@
...
+                        "value": "0"
+                    },
+                    {
+                        "label": "PF2E.SpecificRule.CrystalLuminescence.Enhanced",
+                        "value": "10"
+                    }
+                ],
+                "toggleable": true,
+                "value": true
+            }
```
Comment (TikaelSol) [inline packs/feats/crystal-luminescence.json:60]:
```suggestion
            {
                "domain": "all",
                "key": "RollOption",
                "option": "crystal-luminescence",
                "suboptions": [
                    {
                        "label": "PF2E.SpecificRule.CrystalLuminescence.Normal",
                        "value": "0"
                    },
                    {
                        "label": "PF2E.SpecificRule.CrystalLuminescence.Enhanced",
                        "value": "10"
                    }
                ],
                "toggleable": true
            }
```
Don't need preset values in these stored in compendium.

**Comment by TikaelSol** [comment]
One more thing we should change here is that this is something that takes an action, so this maybe works better as an effect, with the choice of color stored on the feat itself and the effect referencing that value (though just a toggle is workable, I tend to dislike toggles since the actions tab could become cluttered with them for some builds).

We also want to account for the special condition of being able to take the feat twice which increases the radius. Doing that is going to take a bit of elbow grease, storing the radius on the actor with AE likes and adding to it with the toggle then reading that value from the light effect.

File: packs/feats/crystal-luminescence.json (hunk 3 above)
Comment (farling42) [reply packs/feats/crystal-luminescence.json]: Colour selection added by https://github.com/foundryvtt/pf2e/pull/15116/commits/4e936841d4ade577aa2d7612a36fb9bc73caaa4b

**Comment by farling42** [comment]
> We also want to account for the special condition of being able to take the feat twice which increases the radius. Doing that is going to take a bit of elbow grease, storing the radius on the actor with AE likes and adding to it with the toggle then reading that value from the light effect.

Implemented by https://github.com/foundryvtt/pf2e/pull/15116/commits/b4bc5529d63503ef19cddeb073bc66415fa1f2f3

**Comment by farling42** [comment]
One possible remaining issue is that the ChoiceSet is offered each time that the feat is taken, rather than only being offered the first time that the feat is taken.

**Comment by TikaelSol** [comment]
I would say chosen once, but dancing around that will be difficult.

**Comment by farling42** [comment]
> I would say chosen once, but dancing around that will be difficult.

I spent the morning trying to work out how to do that with rule elements, but failed.

**Comment by TikaelSol** [comment]
Yeah, it's fine if it prompts on both times taken, actually fixing that would be a mess and this is a niche use of a niche feat for a niche ancestry. Only so much we can do there :D

**Pull request merged**